controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    jt_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    lift_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_state_broadcaster:
  ros__parameters:
    joints:
      - ur10e_shoulder_pan_joint
      - ur10e_shoulder_lift_joint
      - ur10e_elbow_joint
      - ur10e_wrist_1_joint
      - ur10e_wrist_2_joint
      - ur10e_wrist_3_joint
      # - lift_joint

jt_controller:
  ros__parameters:
    joints:
      - ur10e_shoulder_pan_joint
      - ur10e_shoulder_lift_joint
      - ur10e_elbow_joint
      - ur10e_wrist_1_joint
      - ur10e_wrist_2_joint
      - ur10e_wrist_3_joint

    command_interfaces:
      - position

    state_interfaces:
      - position
      - velocity

    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.6
      ur10e_shoulder_pan_joint:
        trajectory: 0.1
        goal: 0.1
      ur10e_shoulder_lift_joint:
        trajectory: 0.1
        goal: 0.1
      ur10e_elbow_joint:
        trajectory: 0.1
        goal: 0.1
      ur10e_wrist_1_joint:
        trajectory: 0.1
        goal: 0.1
      ur10e_wrist_2_joint:
        trajectory: 0.1
        goal: 0.1
      ur10e_wrist_3_joint:
        trajectory: 0.1
        goal: 0.1

lift_controller:
  ros__parameters:
    joints:
      - lift_joint

    command_interfaces:
      - position

    state_interfaces:
      - position
      - velocity

    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 2.0
      lift_joint:
        trajectory: 0.05
        goal: 0.05